build.gradle

```diff
@@ -0,0 +1,40 @@
+apply plugin: 'com.android.library'
+
+
+def versionMajor = 1
+def versionMinor = 0
+def versionPatch = 0
+
+android {
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
+
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 29
+        versionCode versionMajor * 10000 + versionMinor * 100 + versionPatch
+        versionName "${versionMajor}.${versionMinor}.${versionPatch}"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles 'consumer-rules.pro'
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
+    implementation 'androidx.appcompat:appcompat:1.0.2'
```

Comment:
why do we need Androidx ? there is no UI component right?

Reply:
cause currently in RetryCallAdapterFactory

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

it still use some annotation in the file.

Reply:
are there no way to remove these? Annotation should be optional. Can you try to remove the annotations in the RetryCallAdapterFactory and see if it still works ?
I am just concerned if customer did not use AndroidX (legacy apps) although it is unlikely.

Reply:
checked again in utils
import androidx.annotation.StringRes; 

still need to androidx to access resource string.
if customer did not use AndroidX (legacy apps) -> then androidx will still be imported in our sdk, right ? i mean i do not see any difference whether customer use it or not. when package it. it will be added into library anyway.

Reply:
if they do not use, importing our SDK will create a conflict in Intellij and force them to upgrade all their UI components as well as any existing UI dependencies to AndroidX. I had this issue when trying to do the same for our Android SDK. It causes lots of problems. But if it is unavoidable, we can keep, since the direction from Google is to force the AndroidX update.